### ISSUE TITLE
Segmentation fault with cooperative-sticky assignment

### ISSUE THREAD

**Issue by alistairking** [edenhill/librdkafka#3323]
Description
===========
Quickly restarting a consumer when using cooperative-sticky assigment and static consumer groups results in a segmentation fault:

```
./sticky-bug.sh
Run 1
% Waiting for group rebalance
% Group sticky-assign-group-1617058031 rebalanced: incremental assignment of 1 partition(s) (memberid sticky-test-1617058033059, COOPERATIVE rebalance protocol): sticky-assign-topic [0]
TEST-MSG 1
% Group sticky-assign-group-1617058031 rebalanced: incremental revoke of 1 partition(s) (memberid sticky-test-1617058033059, COOPERATIVE rebalance protocol): sticky-assign-topic [0]
sleeping 1.5s
Run 2
% Waiting for group rebalance
% Group sticky-assign-group-1617058031 rebalanced: incremental assignment of 1 partition(s) (memberid sticky-test-1617058033063, COOPERATIVE rebalance protocol): sticky-assign-topic [0]
TEST-MSG 2
% Group sticky-assign-group-1617058031 rebalanced: incremental revoke of 1 partition(s) (memberid sticky-test-1617058033063, COOPERATIVE rebalance protocol): sticky-assign-topic [0]
sleeping 1.5s
Run 3
% Waiting for group rebalance
./sticky-bug.sh: line 17: 344933 Segmentation fault      $CMD
```

Stack trace from gdb:
```
Thread 10 "rdk:main" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffbf239700 (LWP 337187)]
balance (revocationRequired=0 '\000', currentPartitionConsumer=0x7fffbf234db0, partition2AllPotentialConsumers=0x7fffbf234cf0, consumer2AllPotentialPartitions=0x7fffbf234d50,
    sortedCurrentSubscriptions=0x7fffbf234ba0, unassignedPartitions=0x7fffb00099a0, sortedPartitions=0x7fffb0000f80, prevAssignment=0x7fffbf234c90, currentAssignment=0x7fffbf234c30,
    partitionMovements=0x7fffbf234f30, rk=0x7fffc8002a50) at rdkafka_sticky_assignor.c:980
980                              sortedCurrentSubscriptions))->key)->cnt == 0;
(gdb) bt
#0  balance (revocationRequired=0 '\000', currentPartitionConsumer=0x7fffbf234db0, partition2AllPotentialConsumers=0x7fffbf234cf0, consumer2AllPotentialPartitions=0x7fffbf234d50,
    sortedCurrentSubscriptions=0x7fffbf234ba0, unassignedPartitions=0x7fffb00099a0, sortedPartitions=0x7fffb0000f80, prevAssignment=0x7fffbf234c90, currentAssignment=0x7fffbf234c30,
    partitionMovements=0x7fffbf234f30, rk=0x7fffc8002a50) at rdkafka_sticky_assignor.c:980
#1  rd_kafka_sticky_assignor_assign_cb (rk=0x7fffc8002a50, rkas=<optimized out>, member_id=<optimized out>, metadata=<optimized out>, members=0x7fffb0000f60, member_cnt=0,
    eligible_topics=0x7fffb0000c60, eligible_topic_cnt=0, errstr=0x7fffbf235190 "", errstr_size=512, opaque=0x0) at rdkafka_sticky_assignor.c:1885
#2  0x00007ffff7ab6484 in rd_kafka_assignor_run (rkcg=0x7fffc800a3c0, rkas=0x7fffc8003e50, metadata=0x7fffb0001030, members=members@entry=0x7fffb0000f60, member_cnt=member_cnt@entry=0,
    errstr=errstr@entry=0x7fffbf235190 "", errstr_size=512) at rdkafka_assignor.c:364
#3  0x00007ffff7a9725a in rd_kafka_cgrp_assignor_run (rkcg=0x7fffc800a3c0, rkas=0x7fffc8003e50, err=<optimized out>, metadata=<optimized out>, members=0x7fffb0000f60, member_cnt=0)
    at rdkafka_cgrp.c:1436
#4  0x00007ffff7a977ff in rd_kafka_cgrp_assignor_handle_Metadata_op (rk=<optimized out>, rkq=<optimized out>, rko=<optimized out>) at rdkafka_cgrp.c:1508
#5  0x00007ffff7a776f4 in rd_kafka_op_call (rk=<optimized out>, rkq=<optimized out>, rko=0x7fffb00099c0) at rdkafka_op.c:687
#6  0x00007ffff7a78568 in rd_kafka_op_handle (rk=0x7fffc8002a50, rkq=0x7fffbf235480, rko=0x7fffb00099c0, cb_type=RD_KAFKA_Q_CB_CALLBACK, opaque=0x7fffc800a3c0,
    callback=0x7ffff7aa2bc0 <rd_kafka_cgrp_op_serve>) at rdkafka_op.c:856
#7  0x00007ffff7a70ae0 in rd_kafka_q_serve (rkq=0x7fffc8003ad0, timeout_ms=<optimized out>, max_cnt=max_cnt@entry=0, cb_type=cb_type@entry=RD_KAFKA_Q_CB_CALLBACK, callback=callback@entry=0x0,
    opaque=opaque@entry=0x0) at rdkafka_queue.c:497
#8  0x00007ffff7a37d04 in rd_kafka_thread_main (arg=arg@entry=0x7fffc8002a50) at rdkafka.c:2006
#9  0x00007ffff7ac2f07 in _thrd_wrapper_function (aArg=<optimized out>) at tinycthread.c:576
#10 0x00007ffff78024a4 in start_thread (arg=0x7fffbf239700) at pthread_create.c:456
#11 0x00007ffff7544d0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
```

How to reproduce
================
I can reliably reproduce this with the following trivial script (kafkacat master (@`2ba6861dd043fe13bf5188aefca10b0ca746d60c`) and librdkafka v1.6.1)

```
#!/usr/bin/env bash

set -e

BROKER=localhost:9192
TOPIC=sticky-assign-topic
GROUP=sticky-assign-group-`date +%s`

# produce some test messages to a topic
for X in `seq 10`; do
        echo "TEST-MSG $X" | kafkacat -P -b $BROKER -t $TOPIC -p 0
done

CMD="kafkacat -G $GROUP -b $BROKER -X auto.offset.reset=earliest -X partition.assignment.strategy=cooperative-sticky -X session.timeout.ms=30000 -X group.instance.id=sticky-test -c 1 $TOPIC"

# run the consumer three times in quick-succession (but not too quick)
for X in `seq 4`; do
        echo "Run $X"
        $CMD
        echo "sleeping 1.5s"
        sleep 1.5
done
```

Checklist
=========

 - [x] librdkafka version (release number or git tag): `v1.6.1`
 - [x] Apache Kafka version: `2.3.0`
 - [x] librdkafka client configuration: `auto.offset.reset=earliest,partition.assignment.strategy=cooperative-sticky,session.timeout.ms=30000,group.instance.id=sticky-test `
 - [x] Operating system: `Debian GNU/Linux 9.13 (stretch)`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
Thanks for a great report!

This has been fixed in #3235 and will be available in the upcoming 1.7.0 release (end of april).

**Comment by alistairking** [comment]
Oh sorry, should've tried this against latest master.

Thanks for all the hard work, love the library!
